pkg/controlplane/agent_grpc_execution_updates.go

```diff
@@ -81,18 +81,42 @@ func (s *Server) getExecutionUpdates(ctx context.Context, log *zap.SugaredLogger
 	return updates
 }
 
-func (s *Server) getNextExecution(ctx context.Context, log *zap.SugaredLogger, info scheduling.RunnerInfo) []*executionv1.ExecutionStart {
-	// Run THE schedule query that will atomically assign one or no execution to this runner.
-	exe, ok, err := s.scheduler.ScheduleExecution(ctx, info)
-	if err != nil {
-		log.Warnw("error scheduling execution", "err", err)
-		return nil
+func (s *Server) getNextExecutions(ctx context.Context, log *zap.SugaredLogger, info scheduling.RunnerInfo) []*executionv1.ExecutionStart {
```

Comment:
Polling so often becomes much less important since it can start all requested executions at once. The poll interval should still be low enough to respond promptly to transitioning events (e.g. pausing).